### DISCUSSION TITLE
Prevent non-spawnable objects to be included in DefaultPrefabObjects

### DISCUSSION THREAD

**Discussion by FREEZX** [FirstGearGames/FishNet#518] [Feature Request]
Continuing on the [discussion on discord](https://discord.com/channels/424284635074134018/923983677980045382/1167434628227940433)

NetworkObjects that exist solely as scene objects or that are part of other spawnable prefabs should not be included in the DefaultPrefabObjects, as all references to GameObject assets force unity to store the prefab and all related data in memory for fast instantiation when needed. An easy solution to this would be to add an "IsSpawnable" field to the NetworkObject script, and only take objects with this field set to true when building the DefaultPrefabObjects script. Optionally, for backwards compatibility, this feature should be a checkbox in the FishNet settings.

**Comment by FirstGearGames** [comment]
This has been added.
